hw11-weak-hashmap/src/main/java/ru/otus/cachehw/MyCache.java

```diff
@@ -0,0 +1,53 @@
+package ru.otus.cachehw;
+
+import java.util.HashSet;
+import java.util.WeakHashMap;
+
+/**
+ * @author sergey
+ * @author Korepanov D.
+ * created on 14.12.18.
+ */
+public class MyCache<K, V> implements HwCache<K, V> {
+
+    private WeakHashMap<K,V> cacheMap = new WeakHashMap();
+    private HashSet<HwListener> listeners = new HashSet<HwListener>();
```

Comment:
Лучше переменные объявлять как интерфейсы, а не как конкретные реализации.
Проще при необходимости рефакторить.

```suggestion
    private Map<K,V> cacheMap = new WeakHashMap();
    private Set<HwListener> listeners = new HashSet<HwListener>();
```